cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -12,6 +12,102 @@ WORKER_NODE_ROOT="${CLUSTER_NAME}-worker"
 
 OPERATOR_GIT_HASH=8d3c30de8ec5a9a0c9eeb84ea0aa16ba2395cd68  # release-4.4
 
+# This function gets a command string and invoke it
+# until the command returns an empty string or until timeout
+function retry {
+  local -r tries=$1
+  local -r wait_time=$2
+  local -r action=$3
+  local -r wait_message=$4
+
+  local result=$(eval $action)
```

Comment:
I think this needs to be in the for loop:
```
for i in ..; do
    result=$(...)
    if [ -z result ] ...
        echo ...
        sleep ...
    else  // probably swapping by negating the condition will make more sense.
        break
    fi
done
return result
```

Reply:
Consider this example:
`retry "3" "1" "kubectl get po -l app=app" "trying to get pods with label app=app" && echo "success"`

If there are no pods with the label `app=app` the kubectl output would be "No resource found" without raising any error,
then instead of terminating the loop continues and eventually the output would be "success".